src/Features/Core/Portable/CodeCleanup/AbstractCodeCleanupService.cs

```diff
@@ -136,7 +156,92 @@ protected AbstractCodeCleanupService(ICodeFixService codeFixService)
             return document;
         }
 
+        private async Task<Document> ApplyThirdPartyCodeFixesAsync(Document document, IProgressTracker progressTracker, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var range = new TextSpan(0, tree.Length);
+
+            // Compute diagnostics for everything that is not an IDE analyzer
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range,
+                shouldIncludeDiagnostic: static diagnosticId => !(IDEDiagnosticIdToOptionMappingHelper.IsKnownIDEDiagnosticId(diagnosticId) || IsCompilerDiagnostic(diagnosticId)),
+                includeSuppressedDiagnostics: false,
+                cancellationToken: cancellationToken).ConfigureAwait(false));
+
+            // ensure more than just known diagnostics were returned
+            if (!diagnostics.Any())
+            {
+                return document;
+            }
+
+            var distinctDiagnostics = diagnostics.SelectAsArray(static d => d.Id).Distinct();
+
+            // ensure progress is reported for each third party code fix
+            progressTracker.AddItems(distinctDiagnostics.Length);
+            foreach (var diagnosticId in distinctDiagnostics)
+            {
+                using (progressTracker.ItemCompletedScope(diagnosticId))
+                {
+                    // Apply codefixes for diagnostics with a severity of warning or higher
+                    var updatedDocument = await _codeFixService.ApplyCodeFixesForSpecificDiagnosticIdAsync(
+                        document, diagnosticId, DiagnosticSeverity.Warning, progressTracker, options, cancellationToken).ConfigureAwait(false);
```

Comment:
Do we have plans to make the minimum severity configurable?

Reply:
yeah, going to add a better UI where this can be configured.

Reply:
in this PR?  or in a follow up?

Reply:
follow up, code cleanup UI is currently not extensible in this way and I need to make platform changes